File: backend/OrganizedLife.API/Models/Entities/Transaction.cs
namespace OrganizedLife.API.Models.Entities;

public enum TransactionType
{
    Income,
    Expense
}

public class Transaction
{
    public int Id { get; set; }
    public decimal Amount { get; set; }
    public string Description { get; set; } = string.Empty;
    public TransactionType Type { get; set; }
    public DateTime Date { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Foreign keys
    public int UserId { get; set; }
    public int? CategoryId { get; set; }

    // Navigation properties
    public User User { get; set; } = null!;
    public Category? Category { get; set; }
}

File: frontend/src/types/transaction.ts
export enum TransactionType {
  INCOME = 'income',
  EXPENSE = 'expense',
}

export interface Transaction {
  id: number,
  description: string,
  amount: number,
  type: TransactionType,
  date: string,
  categoryId: number,
  categoryName: string,
  categoryColor: string,
  categoryIcon: string,
  createdAt: string,
}

export interface TransactionSummary {
  totalIncome: number,
  totalExpense: number,
  balance: number,
  transactionCount: number,
}

export interface CreateTransactionDTO {
  description: string,
  amount: number,
  type: TransactionType,
  date: string,
  categoryId: number,
}

export interface UpdateTransactionDTO extends CreateTransactionDTO {
  id: number,
}
